lib/SILGen/SILGen.cpp

```diff
@@ -719,6 +719,26 @@ static bool isEmittedOnDemand(SILModule &M, SILDeclRef constant) {
   }
   case SILDeclRef::Kind::EnumElement:
     return true;
+  case SILDeclRef::Kind::DefaultArgGenerator: {
+    // Default arguments of C++ functions are only emitted is used.
+    auto funcDecl = constant.getFuncDecl();
+    if (funcDecl) {
+      // If the function is inherited from a base C++ class, we have synthesized
+      // a new Swift function that dispatches to the base method and has no
+      // clang node: the underlying clang decl is synthesized lazily. This means
+      // we can't rely on `funcDecl->hasClangNode()` here. Instead, check if any
+      // of the parameters has a clang node.
```

Comment:
That feels like something we should record about the synthesized Swift function.

Reply:
Oh, I didn't realize we could just check the function's module context. Fixed.